crates/node/src/lib.rs

```diff
@@ -986,6 +1032,112 @@ impl Node {
+                        application_id: meta.application_id.into_string().into(),
+                    }),
+                    types::CatchupRequest {
+                        context_id,
+                        last_executed_transaction_hash: meta.last_transaction_hash.into(),
+                        batch_size: self.network_client.catchup_config.batch_size,
+                    },
+                ),
+                None => (
+                    None,
+                    types::CatchupRequest {
+                        context_id,
+                        last_executed_transaction_hash: calimero_primitives::hash::Hash::default(),
+                        batch_size: 50,
+                    },
+                ),
+            };
+
+        let mut stream = self.network_client.open_stream(chosen_peer).await?;
+
+        let request = serde_json::to_vec(&types::CatchupStreamMessage::Request(request))?;
+
+        stream
+            .send(calimero_network::stream::Message { data: request })
+            .await?;
+
+        while let Some(message) = stream.next().await {
+            match serde_json::from_slice(&message?.data)? {
+                types::CatchupStreamMessage::Response(response) => {
+                    let Some(ref context) = context else {
+                        eyre::bail!("Received response with transactions before meta")
+                    };
+
+                    for transaction in response.transactions {
+                        match transaction.status {
+                            types::TransactionStatus::Pending => {
+                                self.tx_pool.insert(
+                                    chosen_peer,
+                                    calimero_primitives::transaction::Transaction {
+                                        context_id: context.id,
+                                        method: transaction.transaction.method,
+                                        payload: transaction.transaction.payload,
+                                        prior_hash: transaction.transaction.prior_hash,
+                                    },
+                                    None,
+                                )?;
+                            }
+                            types::TransactionStatus::Executed => {
+                                self.execute_transaction(
+                                    context.clone(),
+                                    transaction.transaction,
+                                    transaction.transaction_hash,
+                                )
+                                .await?;
+                            }
+                        }
+                    }
+                }
+                types::CatchupStreamMessage::ResponseMeta(response) => match context {
+                    Some(ref context) => {
+                        if context.application_id != response.application_id {
+                            eyre::bail!(
```

Comment:
this should be expected, and accounted for.. we should adopt the new application ID and download the blob

Reply:
you can't directly modify the value view.. but you can re-define it in with `handle.put`